### DISCUSSION TITLE
Send message/execute action on login

### DISCUSSION THREAD

**Discussion by ich777** [authelia/authelia#6517] [Ideas]
Hi,

I just was wondering if it is possible to send a message with a POST request or execute a bash script if someone has successful logged in with Authelia.
This would be a great addition in my opinion since you then also see if one somehow gained access to your credentials and was able to log in.

Cheers,
Christoph

**Comment by james-d-elliott** [comment]
We don't currently support this but it's a decent idea to fire event based calls like this if configured by an admin. How would a specific user be given access to the information afterwards or what kind of call would facilitate this specific part?

**Comment by ich777** [comment]
Thank you for the response! :)

I'm just trying to achieve to send a message through [Pushbits](https://www.pushbits.io/) to my Matrix-Synapse server so that it notifies me that someone logged in.
Pushbits basically uses the Gotify API where you can send a message like [that](https://gotify.net/docs/pushmsg).

I mean there are so many different kinds of notifications agents nowadays but nearly all use a similar API call to send a message for example Pushover [here](https://pushover.net/api).

There are also projects like [Apprise](https://github.com/caronc/apprise) that try to support nearly every notification service through their app so it could get pretty difficult quickly, my main issue is that I've not gone through the code or how Authelia works behind the curtains and I really don't know what's possible.

Maybe you have some ideas... Even the execution from a bash script would be enough. I also could set up a log reader and watch for certain messages that Authelia gives and send notifications on that but I would be really nice to have that built in.

**Reply by james-d-elliott** [reply]
Yeah my main concerns are around security. I don't think an end-user should be able to configure this behaviour to any large extent (maybe just a username or identifier for the service) but wondering what methodology there is to getting it to the user who's logging in. If the idea is just for it to go to admins for all users then that should be easy enough to do in an understandable way. 

Were you hoping for end-user notification or just admin notification?

**Reply by ich777** [reply]
I agree that end users should not be able to configure that.

Just admins should be allowed to do that and set that up.
I think admin notifications should be fine and users should not be able to configure that for themselfs at all. The only thing that I could imagine is that a user could ask a admin to set that up for him, however admin notifications should be enough for most people (including me).

**Comment by LucasWhitmanWasTaken** [comment]
I think this would be a great idea as well, for me personally I would use this to send a request to a discord web-hook, it would be cool if upon authentication authelia would pass some variables to the script.

For example: (Upon an authentication event authelia could provide the following variables.)

- Type of event (Login, logout, timed out for to many attempts, server error, etc.)
- User.
- Remote host details. (IP or other forms of ID)
- Application Accessed.
---
I also agree on the matter that users should not be able to have any configuration of the bash script. I think it should be as simple as authelia executing a script placed by a administrator in a certain directory.

###### Thank you. Have a great day :D

**Reply by james-d-elliott** [reply]
We have settled on the idea if we add anything like this it'll be done via implementing a web hook.

**Reply by LucasWhitmanWasTaken** [reply]
Oh this is great thanks. Is there any definitive plan to add a feature like this? 😀

**Reply by harsh-side** [reply]
> We have settled on the idea if we add anything like this it'll be done via implementing a web hook.

wonderful! 
Do we have timeline for this feature implementation?

**Reply by rabid-dev** [reply]
I also want to know this
